### PR TITLE
Add queue and stack helper functions

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add queue and stack helper functions.

Queue and stack have unique helper functions for inserting and removing map entries.

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: tests/sample/map.c [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        bpf_printk("bpf_map_push_elem returned %d", result);
+        return result;
+    }
+
+    result = bpf_map_peek_elem(map, &value);
+    if (result < 0) {
+        bpf_printk("bpf_map_peek_elem returned %d", result);
+        return result;
+    }
+
```
Comment (dthaler) [inline tests/sample/map.c:91]: This doesn't actually test that the value returned is the value pushed.
Also it doesn't test order.  Please add tests.
E.g., for stack:
push 1
push 2
pop (should be 2)
pop (should be 1)

**Review by poornagmsft — APPROVED** [review]

File: libs/execution_context/unit/execution_context_unit_test.cpp [hunk 2]
```diff
@@ -483,41 +486,38 @@ TEST_CASE("map_crud_operations_stack", "[execution_context]")
...
-            map.get(), 0, NULL, sizeof(extra_value), reinterpret_cast<uint8_t*>(&extra_value), EBPF_ANY, 0) ==
+        ebpf_map_push_entry(map.get(), sizeof(extra_value), reinterpret_cast<uint8_t*>(&extra_value), 0) ==
         EBPF_OUT_OF_SPACE);
 
+    // Replace the oldest entry.
+    REQUIRE(
+        ebpf_map_push_entry(map.get(), sizeof(extra_value), reinterpret_cast<uint8_t*>(&extra_value), BPF_EXIST) ==
+        EBPF_SUCCESS);
+
     // Peek the first element.
```
Comment (dthaler) [inline libs/execution_context/unit/execution_context_unit_test.cpp:505]:
```suggestion
    // Peek at the first element.
```

File: libs/execution_context/unit/execution_context_unit_test.cpp [hunk 3]
```diff
@@ -430,39 +430,42 @@ TEST_CASE("map_crud_operations_queue", "[execution_context]")
...
```
Comment (dthaler) [inline libs/execution_context/unit/execution_context_unit_test.cpp:453]:
```suggestion
    // Peek at the first element.
```

File: tests/sample/map.c [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    PUSH_VALUE(map, 10, TRUE, 0);
+
+    PEEK_VALUE(map, (map == &STACK_map) ? 10 : 1, 0);
+
+    for (i = 0; i < 10; i++) {
+        POP_VALUE(map, (map == &STACK_map) ? 10 - i : i + 1, 0);
+    }
+
+    PEEK_VALUE(map, 0, -7);
+
```
Comment (dthaler) [inline tests/sample/map.c:135]: Also add a test for `POP_VALUE` when the stack/queue is empty, here and by 119, to verify a negative value is returned by bpf_map_pop_value() too.

**Review by dthaler — APPROVED** [review]
